pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -104,10 +105,12 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 	hllSize := toputils.BytesToUint16LittleEndian(fdata[idx : idx+2])
 	idx += 2
 
-	err := sst.CreateHllFromBytes(fdata[idx:idx+hllSize], false)
-	if err != nil {
-		log.Errorf("qid=%d, readSingleSst: unable to create Hll from raw bytes. sst err: %v", qid, err)
-		return nil, err
+	if version == utils.VERSION_SEGSTATS[0] {
+		err := sst.CreateHllFromBytes(fdata[idx : idx+hllSize])
+		if err != nil {
+			log.Errorf("qid=%d, readSingleSst: unable to create Hll from raw bytes. sst err: %v", qid, err)
+			return nil, err
+		}
 	}
```

Comment:
What do you think about something like this instead?
```suggestion
	switch version {
	case 2:
		err := sst.CreateHllFromBytes(fdata[idx : idx+hllSize])
		if err != nil {
			log.Errorf("qid=%d, readSingleSst: unable to create Hll from raw bytes. sst err: %v", qid, err)
			return nil, err
		}
	case 1:
		log.Infof("qid=%d, readSingleSst: ignoring HLL (old version)", qid)
	default:
		// log and return an error
	}
```